subgenerator/templates/index.js

```diff
@@ -4,6 +4,21 @@ var chalk = require('chalk');
 var yosay = require('yosay');
 
 module.exports = yeoman.Base.extend({
+
+  // Note: arguments and options should be defined in the constructor.
+  constructor: function () {
+    yeoman.Base.apply(this, arguments);
+
+    // This makes 'arg' an optional argument.
```

Comment:
I can easily remove the comments. Which lines are unnecessary? The examples of reading arguments and options? Other stuff too? Technically we don't need to have an option or argument registered at all, could completely remove the constructor... Give me some more direction and I'll make it happen.